src/main/java/seedu/mtracker/console/AddEtfParser.java

```diff
@@ -2,11 +2,56 @@
 
 import seedu.mtracker.commands.AddEtfCommand;
 import seedu.mtracker.commands.AddInstrumentCommand;
+import seedu.mtracker.error.ErrorMessage;
+import seedu.mtracker.ui.TextUi;
 
 public class AddEtfParser extends AddInstrumentParser {
 
+    public static String INSTRUMENT_TYPE = "etf";
+
+    public String getEtfRemarksFromUser() {
+        TextUi.displayAddRemarksInstruction();
+        return getUserInput();
+    }
+
+    public String getEtfpastReturnsFromUser() {
+        TextUi.displayAddPastReturnsInstruction();
+        return getUserInput();
+    }
+
+    public static boolean isValidPastReturn(String pastReturns) {
+        boolean isValid = true;
+        try {
+            Double.parseDouble(pastReturns);
+        } catch (NumberFormatException e) {
+            ErrorMessage.displayAddInstrumentCurrentPriceError();
+            isValid = false;
+        }
+        return isValid;
+    }

```

Comment:
There is no need to redeclare cause AddInstrumentParser already has it. Do consider rename ErrorMessage.displayAddPriceError instead of currentPrice cause its being used in more than one place

Reply:
However if its optional may need to create a new checker that first check if input is empty, if its not empty than check if is validPrice

Reply:
I deleted this whole part in the end, cause past returns is optional. Instead if the input cannot be parsed as a double, it will be  assumed as undefined which is default to value of 0. Meaning if left empty, past returns will be assumed as 0.